languages/csharp/exercises/concept/exceptions/.meta/Example.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+static class SimpleCalculator
+{
+    public static string Calculate(int operand1, int operand2, string operation)
+    {
+        int result = 0;
+        try
+        {
+            switch(operation)
+            {
+                case "+":
+                    result = Addition(operand1, operand2);
+                    break;
+                case "*":
+                    result = Multiplication(operand1, operand2);
+                    break;
+                case "/":
+                    result = Division(operand1, operand2);
+                    break;
+                case "":
+                    throw new ArgumentException("Operation cannot be empty.");
+                case null:
+                    throw new ArgumentNullException("Operation cannot be null.");
+                default:
+                    throw new ArgumentOutOfRangeException(operation, $"Operation {operation} does not exist");
+            }
+        }
+        catch(OverflowException e)
```

Comment:
I really like the way this looks now! Some things that we can consider doing:

1. Add the parameter name as an argument to the `ArgumentException`, `ArgumentNullException` and `ArgumentOutOfRangeException`.
1. Instead of catching the `OverflowException` and then throwing a different exception, maybe we could return a string with a special overflow message? This would match how the `user-defined-exceptions` exercise is structured: https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/user-defined-exceptions/.meta/Example.cs#L33 If you like this, we should probably change the theme a little bit to also match that of the `user-defined-exceptions` exercise. Exercises that build on each other are something that we feel is really useful!
- Catch the `DivideByNullException` and return a special string (see above comment)
- The `switch` statement is nice, but it does mean taking a dependency on the `switch` concept. Maybe instead of using a `switch`, we could just use a regular `if` statement. I'm not entirely sure about this though. @mikedamay what do you thing?

Reply:
1. I've adjusted the parameters for those 3 exceptions.
2. We return strings that contain error descriptions for those 2, so far I've implemented  $"The result of operation {operand1} {operation} {operand2} does not fit into integer type." for OverflowException and  "Division by zero is not allowed." for DivideByZero in their respecive catch clauses.

Concerning the switch, I am ok with refactoring it to if blocks if that fits better into the progression tree. Just let me know and I will make the change and repost the solution.

Reply:
@mikedamay What do you think? Should we replace the `switch` with regular `if` statements here?

Reply:
Sorry for missing this.   I tend to favour the fewest number of dependencies so for me `if` is better.  We actually throw an exception in the `switch-statement` exercise which is a natural pattern.  If you're OK with it @archrisV `if`s would be great.

Reply:
Thanks for your input, I will refactor the solution to use the `if` and post it later today.